ResourceStatusSystem/Agent/EmailAgent.py

```diff
@@ -6,44 +6,47 @@
 
 import os
 import sqlite3
-from DIRAC                                                       import gConfig, S_OK, S_ERROR
-from DIRAC.Core.Base.AgentModule                                 import AgentModule
-from DIRAC.ResourceStatusSystem.Utilities                        import RssConfiguration
-from DIRAC.Interfaces.API.DiracAdmin                             import DiracAdmin
+from DIRAC import gConfig, S_OK, S_ERROR
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.ResourceStatusSystem.Utilities import RssConfiguration
+from DIRAC.Interfaces.API.DiracAdmin import DiracAdmin
 
-__RCSID__ = '$Id: $'
+__RCSID__ = '$Id$'
 
 AGENT_NAME = 'ResourceStatus/EmailAgent'
 
-class EmailAgent( AgentModule ):
 
-  def __init__( self, *args, **kwargs ):
+class EmailAgent(AgentModule):
 
-    AgentModule.__init__( self, *args, **kwargs )
+  def __init__(self, *args, **kwargs):
+
+    AgentModule.__init__(self, *args, **kwargs)
     self.diracAdmin = None
     self.default_value = None
 
     if 'DIRAC' in os.environ:
-      self.cacheFile = os.path.join( os.getenv('DIRAC'), 'work/ResourceStatus/cache.db' )
+      self.cacheFile = os.path.join(os.getenv('DIRAC'), 'work/ResourceStatus/cache.db')
     else:
       self.cacheFile = os.path.realpath('cache.db')
 
-  def initialize( self ):
+  def initialize(self):
     ''' EmailAgent initialization
     '''
 
     self.diracAdmin = DiracAdmin()
 
     return S_OK()
 
-  def execute( self ):
+  def execute(self):
 
     if os.path.isfile(self.cacheFile):
       with sqlite3.connect(self.cacheFile) as conn:
 
         result = conn.execute("SELECT DISTINCT SiteName from ResourceStatusCache;")
         for site in result:
-          cursor = conn.execute("SELECT StatusType, ResourceName, Status, Time, PreviousStatus from ResourceStatusCache WHERE SiteName='"+ site[0] +"';")
+          query = "SELECT StatusType, ResourceName, Status, Time, PreviousStatus from ResourceStatusCache "
+          query += "WHERE SiteName='%s';" % site[0]
```

Comment:
escape the query ?

Reply:
This query is safe, there's nothing to escape.